src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java

```diff
@@ -32,150 +35,54 @@
-                command.append(VideoCommand.AUDIO_CODEC.getCommand());
-                command.append(space);
-                command.append(videoParam.getVideoCodec());
-            }
-
-            if (!videoParam.getAudioCodec().equals("")) {
-                command.append(space);
-                command.append(VideoCommand.AUDIO_CODEC.getCommand());
-                command.append(space);
-                command.append(videoParam.getAudioCodec());
-            }
-
-            if (!videoParam.getVideoCodec().equals("")) {
-                command.append(space);
-                command.append(VideoCommand.VIDEO_BITRATE.getCommand());
-                command.append(space);
-                command.append(videoParam.getVideoBitRate());
-            }
-
-            if (!videoParam.getAudioBitRate().equals("")) {
-                command.append(space);
-                command.append(VideoCommand.AUDIO_BITRATE.getCommand());
-                command.append(space);
-                command.append(videoParam.getAudioBitRate());
-            }
-
-            if (videoParam.getQuality() > -1) {
-                command.append(space);
-                command.append(VideoCommand.SCALE.getCommand());
-                command.append(space);
-                command.append(videoParam.getQuality());
-                command.append(VideoCommand.EMPTY.getCommand());
-            }
-
-            if (videoParam.getChannelsNumber() > 0) {
-                command.append(space);
-                command.append(VideoCommand.CHANNELS.getCommand());
-                command.append(space);
-                command.append(videoParam.getChannelsNumber());
-                command.append(VideoCommand.EMPTY.getCommand());
-            }
-
-            if (!videoParam.getVolume().equals("")) {
-                command.append(space);
-                command.append(VideoCommand.VOLUME.getCommand());
-                command.append(space);
-                command.append(videoParam.getVolume());
-            }
-
-            if (!videoParam.getRotate().equals("")) {
-                command.append(space);
-                command.append(VideoCommand.ROTATE.getCommand());
-                command.append(space);
-                command.append(videoParam.getRotate());
-            }
+            List<ICommandStrategy> list = new ArrayList<>();
+            list.add(new CommandFFMpegPath());
+            list.add(new CommandInputFilePath(videoParam.getInputPathFile()));
+            list.add(new CommandVideoAspectRatio(Integer.toString(videoParam.getAspectRatio())));
+            list.add(new CommandVideoScale(videoParam.getWidth(), videoParam.getHeight()));
+            list.add(new CommandVideoConverter());
+            list.add(new CommandVideoThumbNail(Integer.parseInt(videoParam.getThumbnail())));
+            list.add(new CommandVideoRotate(videoParam.getRotate()));
+            list.add(new CommandVideoFrameRate(videoParam.getFrameRate()));
```

Comment:
order the list according to the command structure for example: cmd>ffmpeg inputfile.mp4 -ar 20 outputfilename.avi